plantcv/parallel/job_builder.py

```diff
@@ -50,64 +39,64 @@ def job_builder(meta, valid_meta, workflow, job_dir, out_dir, coprocess=None, ot
     for img in images:
         # Create JSON templates for each image
-        img_meta = {"metadata": deepcopy(valid_meta), "observations": {}}
-        coimg_meta = {"metadata": deepcopy(valid_meta), "observations": {}}
+        img_meta = {"metadata": deepcopy(config.metadata_terms), "observations": {}}
+        coimg_meta = {"metadata": deepcopy(config.metadata_terms), "observations": {}}
 
         # If there is an image co-processed with the image
-        if (coprocess is not None) and ('coimg' in meta[img]):
+        if (config.coprocess is not None) and ('coimg' in meta[img]):
             # Create an output file to store the co-image processing results and populate with metadata
             coimg = meta[meta[img]['coimg']]
-            coout = open(os.path.join(".", job_dir, meta[img]["coimg"] + ".txt"), 'w')
+            coout = open(os.path.join(config.tmp_dir, meta[img]["coimg"] + ".txt"), 'w')
             # Store metadata in JSON
             coimg_meta["metadata"]["image"] = {
                 "label": "image file",
                 "datatype": "<class 'str'>",
                 "value": os.path.join(coimg['path'], meta[img]['coimg'])
             }
             # Valid metadata
-            for m in list(valid_meta.keys()):
+            for m in list(config.metadata_terms.keys()):
                 coimg_meta["metadata"][m]["value"] = coimg[m]
             json.dump(coimg_meta, coout)
             coout.close()
 
         # Create an output file to store the image processing results and populate with metadata
-        outfile = open(os.path.join(".", job_dir, img + ".txt"), 'w')
+        outfile = open(os.path.join(config.tmp_dir, img + ".txt"), 'w')
         # Store metadata in JSON
         img_meta["metadata"]["image"] = {
                 "label": "image file",
                 "datatype": "<class 'str'>",
                 "value": os.path.join(meta[img]['path'], img)
             }
         # Valid metadata
-        for m in list(valid_meta.keys()):
+        for m in list(config.metadata_terms.keys()):
             img_meta["metadata"][m]["value"] = meta[img][m]
         json.dump(img_meta, outfile)
-
         outfile.close()
 
         # Build job
-        job_parts = ["python", workflow, "--image", os.path.join(meta[img]['path'], img),
-                     "--outdir", out_dir, "--result", os.path.join(job_dir, img) + ".txt"]
+        job_parts = ["python", config.workflow, "--image", meta[img]['path'],
+                     "--outdir", config.output_dir, "--result",
+                     os.path.join(config.tmp_dir, img) + ".txt"]
         # Add job to list
-        if coprocess is not None and ('coimg' in meta[img]):
-            job_parts = job_parts + ["--coresult", os.path.join(job_dir, meta[img]['coimg']) + ".txt"]
-        if writeimg:
+        if config.coprocess is not None and ('coimg' in meta[img]):
+            job_parts = job_parts + ["--coresult", os.path.join(config.tmp_dir, meta[img]['coimg']) + ".txt"]
```

Comment:
will this fail if config.tmp_dir is not None but doesn't exist?

Reply:
When tmp_dir is None, the `tempfile` package I am using automatically uses a system temporary directory I believe. But I think you are right, if you specify tmp_dir as a location that does not exist, `tempfile` tries to make a temp directory in that space and will fail.